parsely/tests.py

```diff
@@ -72,78 +79,97 @@ def test_shares_detail(self):
-        self.p.referrers_meta_detail('Ars Staff', meta="author", _callback=handle)
+            self.assertEqual(res[3].author, "Ars Staff")
+
+        self.p.referrers_meta_detail("Ars Staff", meta="author", _callback=handle)
 
     def test_referrers_meta(self):
         r = self.p.referrers_meta()
         self.assertTrue(r[0].hits > 0)
 
         def handle(res):
             self.assertTrue(res[0].hits > 0)
+
         self.p.referrers_meta(_callback=handle)
 
     def test_referrers(self):
-        r = self.p.referrers(tag='copyright')
+        r = self.p.referrers(tag="copyright")
         self.assertTrue(r[2].hits > 0)
-        self.assertTrue(r[2].ref_type == 'social')
+        self.assertTrue(r[2].ref_type == "social")
 
         def handle(res):
             self.assertTrue(res[2].hits > 0)
-            self.assertTrue(res[2].ref_type == 'social')
-        self.p.referrers(tag='copyright', _callback=handle)
+            self.assertTrue(res[2].ref_type == "social")
+
+        self.p.referrers(tag="copyright", _callback=handle)
 
     def test_meta_detail(self):
-        r = self.p.meta_detail('Tech', aspect="section")
-        self.assertEquals(r[3].section, "Tech")
+        r = self.p.meta_detail("Tech", aspect="section")
+        self.assertEqual(r[3].section, "Tech")
 
         def handle(res):
-            self.assertEquals(res[3].section, "Tech")
-        self.p.meta_detail('Tech', aspect="section", _callback=handle)
+            self.assertEqual(res[3].section, "Tech")
+
+        self.p.meta_detail("Tech", aspect="section", _callback=handle)
 
     def test_post_detail(self):
-        r = self.p.post_detail('https://arstechnica.com/gadgets/2017/09/android-8-0-oreo-thoroughly-reviewed/')
+        r = self.p.post_detail(
+            "https://arstechnica.com/gadgets/2017/09/android-8-0-oreo-thoroughly-reviewed/"
+        )
         self.assertTrue(r.title == "Android 8.0 Oreo, thoroughly reviewed")
         self.assertTrue(r.visitors > 0)
 
         def handle(res):
             self.assertTrue(res.title == "Android 8.0 Oreo, thoroughly reviewed")
             self.assertTrue(res.visitors > 0)
-        self.p.post_detail('https://arstechnica.com/gadgets/2017/09/android-8-0-oreo-thoroughly-reviewed/', _callback=handle)
+
+        self.p.post_detail(
+            "https://arstechnica.com/gadgets/2017/09/android-8-0-oreo-thoroughly-reviewed/",
+            _callback=handle,
+        )
 
     def test_analytics(self):
         r = self.p.analytics(aspect="authors")
         self.assertTrue(r[7].hits > 0)
 
         def handle(res):
             self.assertTrue(res[7].hits > 0)
+
         self.p.analytics(aspect="authors", _callback=handle)
 
     def test_analytics_one_pubdate(self):
         with self.assertRaises(ValueError):
-            self.p.analytics(aspect="authors", pub_start=datetime(2017, 9, 04))#2013, 10, 01))
+            self.p.analytics(
+                aspect="authors", pub_start=datetime(2017, 9, 0o4)
```

Comment:
This `0o4` is weird, should just be `4`

Reply:
Good catch!  `04` isn't valid syntax in Python 3. Not sure why the default assumption is octal though.